src/app/reporting/SynchronizedReportSchedulerImpl.h

```diff
@@ -42,17 +84,80 @@ class SynchronizedReportSchedulerImpl : public ReportSchedulerImpl, public Timer
+     *
+     * @param[in] timeout The timeout to schedule the report.
+     * @param[in] node The node associated to the ReadHandler.
+     * @param[in] now The current system timestamp.
+     *
+     * @return CHIP_ERROR CHIP_NO_ERROR on success, timer related error code otherwise (This can only fail on starting the timer)
+     */
     CHIP_ERROR ScheduleReport(System::Clock::Timeout timeout, ReadHandlerNode * node, const Timestamp & now) override;
     void CancelReport();
 
 private:
     friend class chip::app::reporting::TestReportScheduler;
 
+    /**
+     * @brief Find the highest minimum timestamp possible that still respects the lowest max timestamp and sets it as the common
+     * minimum. If the max timestamp has not been updated and is in the past, or if no min timestamp is lower than the current max
+     * timestamp, this will set now as the common minimum timestamp, thus allowing the report to be sent immediately.
+     *
+     * @param[in] now The current system timestamp, set by the event that triggered the call of this method.
+     *
+     * @return CHIP_ERROR on success or CHIP_ERROR_INVALID_LIST_LENGTH if the list is empty
+     */
     CHIP_ERROR FindNextMinInterval(const Timestamp & now);
+
+    /**
+     * @brief Find the smallest maximum interval possible and set it as the common maximum
+     *
+     * @param[in] now The current system timestamp, set by the event that triggered the call of this method.
+     *
+     * @return CHIP_ERROR on success or CHIP_ERROR_INVALID_LIST_LENGTH if the list is empty
+     */
     CHIP_ERROR FindNextMaxInterval(const Timestamp & now);
+
+    /**
+     *  @brief Calculate the next report timeout for all ReadHandlerNodes
+     *
+     * @param[out] timeout The timeout to calculate.
+     * @param[in] aReadHandlerNode unused, kept to preserve the signature of the base class
+     * @param[in] now The current system timestamp when the event leading to the call of this method happened.
+     *
+     *  The next report timeout is calculated by looping through all the ReadHandlerNodes and finding if any are reportable now
+     *      or at min.
+     *   * If a ReadHandlerNode is reportable now, the timeout is set to 0.
+     *   * If a ReadHandlerNode is reportable at min, the timeout is set to the difference between the Scheduler's  min timestamp
+     *      and the current time.
+     *   * If no ReadHandlerNode is reportable, the timeout is set to the difference between the Scheduler's max timestamp and the
+     *      current time.
+     *
+     * @note Since this method is called after the OnSubscriptionReportSent callback, to avoid an endless reporting loop, Nodes with
+     * the IsEngineRunScheduled flag set are ignored when finding if the Scheduler should report at min, max or now.
+     *
+     * @note If a ReadHandler's report is Chunked, the IsEngineRunScheduled is ignored since we do want to keep rescheduling the
```

Comment:
This part of the documentation is confusing.  Is this discussing an exception to the "avoid an endless loop" bit above?  If so, it would be clearer to just describe what situations we want to reschedule in and what situations not, without this weird "we do X, except no, in case Y we do Z" structure.